src/OrchardCore.Modules/OrchardCore.AuditTrail/Services/DefaultAuditTrailAdminListFilterProvider.cs

```diff
@@ -0,0 +1,202 @@
+                    {
+                        if (!string.IsNullOrEmpty(val) && DateTimeParser.Parser.TryParse(val, out var expression))
+                        {
+                            var context = (AuditTrailQueryContext)ctx;
+                            var clock = context.ServiceProvider.GetRequiredService<IClock>();
+                            var utcNow = clock.UtcNow;
+
+                            var param = Expression.Parameter(typeof(AuditTrailEventIndex));
+                            var field = Expression.Property(param, nameof(AuditTrailEventIndex.CreatedUtc));
+                            var expressionContext = new BuildExpressionContext(utcNow, param, field, typeof(Func<AuditTrailEventIndex, bool>));
+
+                            query.With<AuditTrailEventIndex>((Expression<Func<AuditTrailEventIndex, bool>>)expression.BuildExpression(expressionContext));
+                        }
+
+                        return new ValueTask<IQuery<AuditTrailEvent>>(query);
+                    })
+                    .MapTo<AuditTrailIndexOptions>((val, model) =>
+                    {
+                        model.Date = val;
+                    })
+                    .MapFrom<AuditTrailIndexOptions>((model) =>
+                    {
+                        if (!String.IsNullOrEmpty(model.Date))
+                        {
+                            return (true, model.Date);
+                        }
+                        return (false, String.Empty);
+                    })
+                )
+                .WithNamedTerm("sort", builder => builder
+                    .OneCondition<AuditTrailEvent>((val, query, ctx) =>
+                    {
+                        var context = (AuditTrailQueryContext)ctx;
+                        var options = context.ServiceProvider.GetRequiredService<IOptions<AuditTrailAdminListOptions>>().Value;
+
+                        if (options.SortOptions.TryGetValue(val, out var sortOption))
+                        {
+                            return sortOption.Query(val, query, ctx);
+                        }
+
+                        return options.DefaultSortOption.Query(val, query, ctx);
+                    })
+                    .MapTo<AuditTrailIndexOptions>((val, model) =>
+                    {
+                        // TODO add a context property to the mapping func.
+                        if (_options.Value.SortOptions.TryGetValue(val, out var sortOption))
+                        {
+                            model.Sort = sortOption.Value;
+                        }
+                    })
+                    .MapFrom<AuditTrailIndexOptions>((model) =>
+                    {
+                        // TODO add a context property to the mapping func.
+                        if (model.Sort != _options.Value.DefaultSortOption.Value)
+                        {
+                            return (true, model.Sort.ToString());
```

Comment:
In testing the filters I got a null object exception here due to model.Sort being null, should there be a null check on this as well as the defaultsortoption check or should the Sort property be handled further back?

Reply:
just should have `.ToString()` called on it. cheers